ds-caselaw-ingester/lambda_function.py

```diff
@@ -369,23 +365,26 @@ def handler(event, context):
         copy_file(
             tar, f"{consignment_reference}/parser.log", "parser.log", uri, s3_client
         )
-    except KeyError:
+    except FileNotFoundException:
         pass
 
     # Store images
-    for image_filename in metadata["parameters"]["TRE"]["payload"]["images"]:
-        copy_file(
-            tar,
-            f"{consignment_reference}/{image_filename}",
-            image_filename,
-            uri,
-            s3_client,
-        )
+    if metadata["parameters"]["TRE"]["payload"]["images"]:
```

Comment:
An alternative might be
`for image_filename in metadata['parameters']['TRE']['payload'].get('images', [])` if `payload` is known to exist.

I'd generally set a variable to that long list of things rather than duplicating it.

Reply:
Thank you for the tip!

Reply:
Run into a snag with this:

```
"parameters": {
    "TRE": {
      "reference": "TRE-TDR-2022-FGD",
      "prod-tre-version": "0.1.2",
      "payload": {
        "filename": "failures_TDR-2022-CRC.docx",
        "xml": null,
        "metadata": "TRE-TDR-2022-FGD-metadata.json",
        "images": null,
        "log": null
      },
```

Using `metadata['parameters']['TRE']['payload'].get('images', [])` results in `NoneType is not iterable`. So `images` exists but is a None 🤔

Reply:
That's a shame; that's fine as is, then. :)